apps/condo/domains/ticket/access/TicketComment.js

```diff
@@ -59,7 +74,29 @@ async function canReadTicketComments ({ authentication: { item: user }, context
     }
 }
 
-const checkManageCommentAccess = async ({ user, operation, originalInput, itemId, context }) => {
+const checkManageCommentAccess = async (args) => {
+    const { authentication: { item: user }, originalInput, operation, itemId, context } = args
+
+    if (user.type === SERVICE) {
+        const hasAccess = await canManageObjectsAsB2BAppServiceUser(args)
+        if (!hasAccess) return false
+
+        // service user can't create ticket comment with resident type or update type to resident
+        const resolvedCommentType = get(originalInput, 'type')
+        if (resolvedCommentType === RESIDENT_COMMENT_TYPE) {
```

Comment:
Broken logic right here, if it's bulk request or there's not type in input, you'll get `undefined` and this check will be passed

Reply:
This code is a bit unclear to read. Since create `return` is in the end, while update logic is mixed somewhere in the middle... How about something like this:

```
if (operation === 'create') {
    const ticketsInput = Array.isArray(originalInput) ? originalInput : [originalInput]
    if (ticketsInput.some(ticketData => ticketData.type && ticketData.type === RESIDENT_COMMENT_TYPE)) {
        return false
    }

    return true
} else if (operation === 'update') {
    const ids = itemIds || [itemId]
    const uniqueIds = uniq(ids)
    const comments = await find('TicketComment', { id_in: uniqueIds, deletedAt: null, user: { id: user.id } })
    
    return comments.length === uniqueIds.length
}

return false
```

Reply:
Example above is more readable imho and support bulk operations (in case we add bulk in canManageAsB2B utils (and we definitely will later))

Reply:
> Broken logic right here, if it's bulk request or there's not type in input, you'll get undefined and this check will be passed

If it bulk request, in `checkManageCommentAccess` function `originalInput` will be `originalInput[i]` from request

`checkManageCommentAccess` calls below in the code in `canManageTicketComments`, which use for access check to `TicketComment` model. Same for TicketCommentFile.

I will only add a check that bulk update is denied (it's not worked before, but not explicit)